app/src/main/res/layout/settings_pill_with_subtitle.xml

```diff
@@ -23,27 +23,35 @@
     android:orientation="vertical"
     android:paddingBottom="10dp">
 
-    <LinearLayout
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:orientation="horizontal">
+        android:layout_height="match_parent">
 
         <androidx.appcompat.widget.AppCompatTextView
             android:id="@+id/title"
             style="@style/SettingsItem"
-            android:layout_width="wrap_content"
-            android:paddingEnd="8dp"
-            tools:text="Title goes here"
-            tools:ignore="RtlSymmetry" />
+            android:layout_width="0dp"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/pill"
+            app:layout_constrainedWidth="true"
+            app:layout_constraintHorizontal_bias="0"
+            app:layout_constraintHorizontal_chainStyle="packed"
+            app:layout_constraintWidth_default="wrap"
+            tools:text="Title goes here" />
 
         <androidx.appcompat.widget.AppCompatImageView
             android:id="@+id/pill"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_gravity="center_vertical"
+            android:layout_marginEnd="16dp"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@id/title"
+            app:layout_constraintTop_toTopOf="parent"
             app:srcCompat="@drawable/ic_beta_pill" />
 
-    </LinearLayout>
+    </androidx.constraintlayout.widget.ConstraintLayout>
 
     <androidx.appcompat.widget.AppCompatTextView
         android:id="@+id/subtitle"
```

Comment:
you can remove constraints from that view

Reply:
Done!